src/interface/gemm_interface.hpp

```diff
@@ -55,6 +55,14 @@ inline typename std::enable_if<is_sycl_scalar<T>::value, bool>::type isZero(
   return (value == static_cast<T>(0));
 }
 
+#ifdef BLAS_ENABLE_HALF
```

Comment:
If you remove the BLAS_ENABLE_HALF guard from is_half in include/blas_meta.h, you can also remove this one.